modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+		}
+
+		$icon_count        = count( $new_instance['url-icons'] );
+		$instance['icons'] = array();
+
+		for ( $i = 0; $i < $icon_count; $i++ ) {
+			$url = filter_var( $new_instance['url-icons'][ $i ], FILTER_SANITIZE_URL );
+
+			if ( ! empty( $url ) ) {
+				$instance['icons'][] = array(
+					'url' => $url,
+				);
+			}
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 *
+	 * @return string|void
+	 */
+	public function form( $instance ) {
+		$instance = wp_parse_args( (array) $instance, $this->defaults );
+		$title    = sanitize_text_field( $instance['title'] );
+		$sizes    = array(
+			'small'  => esc_html__( 'Small', 'jetpack' ),
+			'medium' => esc_html__( 'Medium', 'jetpack' ),
+			'large'  => esc_html__( 'Large', 'jetpack' ),
+		);
+		?>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'title' ); ?>"><?php esc_html_e( 'Title:', 'jetpack' ); ?></label>
+			<input class="widefat" id="<?php echo $this->get_field_id( 'title' ); ?>" name="<?php echo $this->get_field_name( 'title' ); ?>" type="text" value="<?php echo esc_attr( $title ); ?>" />
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'icon-size' ); ?>"><?php esc_html_e( 'Size:', 'jetpack' ); ?></label>
+			<select class="widefat" name="<?php echo $this->get_field_name( 'icon-size' ); ?>">
+				<?php foreach ( $sizes as $value => $label ) : ?>
+					<option value="<?php echo esc_attr( $value ); ?>" <?php selected( $value, $instance['icon-size'] ); ?>><?php echo esc_attr( $label ); ?></option>
+				<?php endforeach; ?>
+			</select>
+		</p>
+
+		<div class="jetpack-social-icons-widget-list"
+			data-url-icon-id="<?php echo $this->get_field_id( 'url-icons' ); ?>"
+			data-url-icon-name="<?php echo $this->get_field_name( 'url-icons' ); ?>"
+		>
+
+			<?php
+				foreach ( $instance['icons'] as $icon ) {
+					self::render_icons_template( array(
+						'url-icon-id'   => $this->get_field_id( 'url-icons' ),
+						'url-icon-name' => $this->get_field_name( 'url-icons' ),
+						'url-value'     => $icon['url'],
+					) );
+				}
+			?>
+
+		</div>
+
+		<p class="jetpack-social-icons-widget add-button">
+			<button type="button" class="button jetpack-social-icons-add-button">
+				<?php esc_html_e( 'Add an icon', 'jetpack' ); ?>
+			</button>
+		</p>
+
+		<?php
+			$support     = 'https://en.support.wordpress.com/widgets/social-media-icons-widget/#available-icons';
+			if ( 'es' === get_locale() ) {
+				$support = 'https://es.support.wordpress.com/social-media-icons-widget/#iconos-disponibles';
+			}
+			if ( 'pt-br' === get_locale() ) {
+				$support = 'https://br.support.wordpress.com/widgets/widget-de-icones-sociais/#ícones-disponíveis';
+			}
+		?>
+
+		<p>
+			<em><a href="<?php echo esc_url( $support ); ?>" target="_blank">
+				<?php esc_html_e( 'View available icons', 'jetpack' ); ?>
+			</a></em>
+		</p>
+
+	<?php
+	}
+
+	/**
```

Comment:
Note to self: This is as far as I got in my review, pick up from here later tonight.